packages/magic-connector/src/index.ts

```diff
@@ -0,0 +1,131 @@
+import { ConnectorUpdate } from '@web3-react/types'
+import { AbstractConnector } from '@web3-react/abstract-connector'
+import invariant from 'tiny-invariant'
+
+type NetworkName = 'mainnet' | 'ropsten' | 'rinkeby' | 'kovan'
+
+const chainIdToNetwork: { [network: number]: NetworkName } = {
+  1: 'mainnet',
+  3: 'ropsten',
+  4: 'rinkeby',
+  42: 'kovan'
+}
+
+interface MagicConnectorArguments {
+  apiKey: string
+  chainId: number
+  email: string
+}
+
+export class UserRejectedRequestError extends Error {
+  public constructor() {
+    super()
+    this.name = this.constructor.name
+    this.message = 'The user rejected the request.'
+  }
+}
+
+export class FailedVerificationError extends Error {
+  public constructor() {
+    super()
+    this.name = this.constructor.name
+    this.message = 'The email verification failed.'
+  }
+}
+
+export class MagicLinkRateLimitError extends Error {
+  public constructor() {
+    super()
+    this.name = this.constructor.name
+    this.message = 'The Magic rate limit has been reached.'
+  }
+}
+
+export class MagicLinkExpiredError extends Error {
+  public constructor() {
+    super()
+    this.name = this.constructor.name
+    this.message = 'The Magic link has expired.'
+  }
+}
+
+export class MagicConnector extends AbstractConnector {
+  private readonly apiKey: string
+  private readonly chainId: number
+  private readonly email: string
+
+  public magic: any
+
+  constructor({ apiKey, chainId, email }: MagicConnectorArguments) {
+    invariant(Object.keys(chainIdToNetwork).includes(chainId.toString()), `Unsupported chainId ${chainId}`)
+    invariant(email && email.includes('@'), `Invalid email: ${email}`)
+    super({ supportedChainIds: [chainId] })
+
+    this.apiKey = apiKey
+    this.chainId = chainId
+    this.email = email
+  }
+
+  public async activate(): Promise<ConnectorUpdate> {
+    const { Magic, RPCError, RPCErrorCode } = await import('magic-sdk')
```

Comment:
does this need the same treatment as in https://github.com/NoahZinsmeister/web3-react/pull/115?

Reply:
i updated this, actually

Reply:
I don’t think this is needed here, since we only use named imports? :thinking:

Reply:
you may be right! out of an abundance of caution though, because i'm no bundling expert, i'm just going to leave my change in. can't really hurt, i don't think

Reply:
Yes I was checking again: https://unpkg.com/browse/magic-sdk@2.4.7/dist/module/index.js

And importing a default here requires some conversion from the bundler, to change `import Magic` into `import * as Magic`. I think named imports should be more compatible − but this is something we could test more to be sure.

Reply:
> you may be right! out of an abundance of caution though, because i'm no bundling expert, i'm just going to leave my change in. can't really hurt, i don't think

Thinking more about it… I don’t think there is any conversion needed: dynamic imports don’t have a static syntax like `import` has, so browsers already convert all the named exports into a single object. Your change might be more safe yes :+1: